tests/trx_generator/trx_provider.hpp

```diff
@@ -42,63 +42,64 @@ namespace eosio::testing {
    using fc::time_point;
 
    struct tps_test_stats {
-      uint32_t total_trxs;
-      uint32_t trxs_left;
-
+      uint32_t total_trxs = 0;
+      uint32_t trxs_left = 0;
+      uint32_t trxs_sent = 0;
       time_point start_time;
       time_point expected_end_time;
-   };
+      time_point last_run;
+      time_point next_run;
+      int64_t last_sleep = 0;
 
-   struct simple_tps_monitor {
-      bool monitor_test(const tps_test_stats& stats) {return true;}
    };
 
-   struct null_trx_generator {
-      void generate_and_send() {}
-   };
 
-   constexpr int64_t min_sleep_us = 100;
+   constexpr int64_t min_sleep_us = 1;
 
    template<typename G, typename M>
    struct trx_tps_tester {
-      G _generator;
-      M _monitor;
+      std::shared_ptr<G> _generator;
+      std::shared_ptr<M>_monitor;
```

Comment:
Why mock out a monitor? Seems like that is what you want to test. Also a space should be added before `_monitor`